packages/core/test/http/requestBuilder.test.ts

```diff
@@ -587,316 +641,360 @@ describe('test default request builder behavior with succesful responses', () =>
     expect(apiResponse.request.url).toEqual(expectedRequestUrl);
   });
-
   it('should test request builder query indexedPrefix parameters with array', async () => {
+    buildRequestWithArrayQuery(
+      indexedPrefix,
+      'https://apimatic.hopto.org:3000/test/requestBuilder?array%5B0%5D=item1&array%5B1%5D=item2'
+    );
+  });
+  it('should test request builder query unindexedPrefix parameters with array', async () => {
+    buildRequestWithArrayQuery(
+      unindexedPrefix,
+      'https://apimatic.hopto.org:3000/test/requestBuilder?array%5B%5D=item1&array%5B%5D=item2'
+    );
+  });
+  it('should test request builder query plainPrefix parameters with array', async () => {
+    buildRequestWithArrayQuery(
+      plainPrefix,
+      'https://apimatic.hopto.org:3000/test/requestBuilder?array=item1&array=item2'
+    );
+  });
+  it('should test request builder query tabPrefix parameters with array', async () => {
+    buildRequestWithArrayQuery(
+      tabPrefix,
+      'https://apimatic.hopto.org:3000/test/requestBuilder?array=item1%09item2'
+    );
+  });
+  it('should test request builder query commaPrefix parameters with array', async () => {
+    buildRequestWithArrayQuery(
+      commaPrefix,
+      'https://apimatic.hopto.org:3000/test/requestBuilder?array=item1%2Citem2'
+    );
+  });
+  it('should test request builder query pipePrefix parameters with array', async () => {
+    buildRequestWithArrayQuery(
+      pipePrefix,
+      'https://apimatic.hopto.org:3000/test/requestBuilder?array=item1%7Citem2'
+    );
+  });
+  it('should test request builder query parameters with complex object', async () => {
     const expectedRequestUrl =
-      'https://apimatic.hopto.org:3000/test/requestBuilder?array%5B0%5D=item1&array%5B1%5D=item2';
+      'https://apimatic.hopto.org:3000/test/requestBuilder?object%5Bkey1%5D=value1&object%5Bkey2%5D=value2&object%5Bkey3%5D%5Bsubkey%5D=12345&object%5Bkey4%5D%5B0%5D=item1&object%5Bkey4%5D%5B1%5D=item2';
 
-    const reqBuilder = buildRequestWithArrayQuery(indexedPrefix);
+    const reqBuilder = defaultRequestBuilder();
+    reqBuilder.query('object', {
+      key1: 'value1',
+      key2: 'value2',
+      key3: {
+        subkey: 12345,
+      },
+      key4: ['item1', 'item2'],
+    });
 
     const apiResponse = await reqBuilder.callAsText();
     expect(apiResponse.request.url).toEqual(expectedRequestUrl);
   });
+  it('should test request builder query with multiple parameters and prefix format', async () => {
+    const reqBuilder = defaultRequestBuilder();
+    reqBuilder.query('array1', ['item1', 'item2'], indexedPrefix);
 
-  it('should test request builder query unindexedPrefix parameters with array', async () => {
+    const prefixFormats = (reqBuilder as any)._queryParamsPrefixFormat;
+    expect(prefixFormats.array1).toBe(indexedPrefix);
+  });
+});
+
+describe('test template function', () => {
+  it('should replace template parameters in path with provided values', async () => {
```

Comment:
All these tests under `test template function` group are pretty identicle.

Please add only 3 tests for these cases:

1. With different types like number, boolean, float, and string.
2. With array types i.e. `/data/{array}` should behave with ['a', 'b', 'c'] like `/data/a/b/c`
3. Verify the similarity of behavior in `appendPath` and `template` functions